### ISSUE TITLE
Error when inserting esp32_spi module

### ISSUE THREAD

**Issue by Oquirella** [espressif/esp-hosted#135]
I get the following errors when inserting the module:

[ 648.322010] Failed to obtain SPI master handle
[ 648.761621] Failed Init SPI device
modprobe: ERROR: could not insert 'esp32_spi': No such device

**Comment by mantriyogesh** [comment]
This error simply means that the SPI device you are capturing is not available for use.
1. Can you please let us know what the Linux system you are using?
2. If it is different from Raspberry Pi, have you configured the Device Tree and verified the device works with user-space driver like spidev?

I would like to take point you to [Porting Guide](https://github.com/espressif/esp-hosted/blob/master/docs/porting_guide.md) for your Linux.
Please do connect if you still have any doubt/issue.

**Comment by Oquirella** [comment]
I am using the imx6ull evalution kit with the following devictree:
https://github.com/Embedfire/ebf-buster-linux/blob/master/arch/arm/boot/dts/imx6ul-14x14-evk-ecspi.dts

/dev/spidev3.0 is available

Any ideas why it is not working?

**Comment by mantriyogesh** [comment]
Can you please refer [this document](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/docs/Linux_based_host/porting_guide.md) and try changing the SPI instance to `3` as per `SPI Bus instance and Chip select number`.

**Comment by mantriyogesh** [comment]
Any way you could progress ahead @Oquirella ?
Do let us know if still face problem.

**Comment by Oquirella** [comment]
I get the following message in my logs:

ESP32 peripheral is registered to SPI bus [0],chip select [0], SPI Clock [10]

But i don't get any device when using ifconfig -a:

can0 Link encap:UNSPEC HWaddr 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00
NOARP MTU:16 Metric:1
RX packets:0 errors:0 dropped:0 overruns:0 frame:0
TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:10
RX bytes:0 (0.0 B) TX bytes:0 (0.0 B)
Interrupt:28

can1 Link encap:UNSPEC HWaddr 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00
NOARP MTU:16 Metric:1
RX packets:0 errors:0 dropped:0 overruns:0 frame:0
TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:10
RX bytes:0 (0.0 B) TX bytes:0 (0.0 B)
Interrupt:29

eth0 Link encap:Ethernet HWaddr 00:04:9F:06:F5:5B
BROADCAST MULTICAST MTU:1500 Metric:1
RX packets:0 errors:0 dropped:0 overruns:0 frame:0
TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:1000
RX bytes:0 (0.0 B) TX bytes:0 (0.0 B)

eth1 Link encap:Ethernet HWaddr 00:04:9F:06:F5:5A
BROADCAST MULTICAST MTU:1500 Metric:1
RX packets:0 errors:0 dropped:0 overruns:0 frame:0
TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:1000
RX bytes:0 (0.0 B) TX bytes:0 (0.0 B)

lo Link encap:Local Loopback
inet addr:127.0.0.1 Mask:255.0.0.0
UP LOOPBACK RUNNING MTU:65536 Metric:1
RX packets:0 errors:0 dropped:0 overruns:0 frame:0
TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:1000
RX bytes:0 (0.0 B) TX bytes:0 (0.0 B)

sit0 Link encap:UNSPEC HWaddr 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00
NOARP MTU:1480 Metric:1
RX packets:0 errors:0 dropped:0 overruns:0 frame:0
TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:1000
RX bytes:0 (0.0 B) TX bytes:0 (0.0 B)

**Comment by mantriyogesh** [comment]
Referring to logs shared at https://github.com/espressif/esp-hosted/issues/137#issuecomment-1216394080

Can you please show the dmesg log?
First INIT msg is expected to be received in the dmesg.

**Comment by Oquirella** [comment]
This is what i get:

[   19.623324] systemd[1]: Condition check resulted in Create list of static device nodes for the current kernel being skipped.
[   19.647385] systemd[1]: Condition check resulted in File System Check on Root Device being skipped.
[   19.691047] systemd[1]: Starting Journal Service...
[   19.740765] systemd[1]: Starting Load Kernel Modules...
[   19.808714] systemd[1]: Starting Remount Root and Kernel File Systems...
[   19.883707] systemd[1]: Starting udev Coldplug all Devices...
[   19.994655] systemd[1]: Mounted POSIX Message Queue File System.
[   20.048596] systemd[1]: Mounted Kernel Debug File System.
[   20.088762] systemd[1]: Mounted Temporary Directory (/tmp).
[   20.142905] systemd[1]: Started Remount Root and Kernel File Systems.
[   20.173138] ESP32 peripheral is registered to SPI bus [0],chip select [0], SPI Clock [10]

I don't get an INIT msg?

**Comment by Oquirella** [comment]
Here is my devicetree:

// SPDX-License-Identifier: (GPL-2.0 OR MIT)
//
// Copyright (C) 2016 Freescale Semiconductor, Inc.

/dts-v1/;

#include "imx6ull.dtsi"
#include "imx6ul-14x14-evk.dtsi"

/ {
	model = "Freescale i.MX6 ULL 14x14 EVK Board";
	compatible = "fsl,imx6ull-14x14-evk", "fsl,imx6ull";
};

&clks {
	assigned-clocks = <&clks IMX6UL_CLK_PLL3_PFD2>, <&clks IMX6UL_CLK_PLL4_AUDIO_DIV>;
	assigned-clock-rates = <320000000>, <786432000>;
};

&iomuxc {
	pinctrl-names = "default";
	pinctrl-0 = <&pinctrl_additionalgpio>;

	pinctrl_ecspi1: ecspi1grp {
		fsl,pins = <
			MX6UL_PAD_LCD_DATA20__ECSPI1_SCLK  0x10b0
			MX6UL_PAD_LCD_DATA21__GPIO3_IO26   0x10b0
			MX6UL_PAD_LCD_DATA22__ECSPI1_MOSI  0x10b0
            MX6UL_PAD_LCD_DATA23__ECSPI1_MISO  0x10b0
		>;
	};

	pinctrl_additionalgpio: additionalgpios {
		fsl,pins = <
			MX6UL_PAD_LCD_VSYNC__GPIO3_IO03 	0x10b0 /* 67 RST */
			MX6UL_PAD_LCD_DATA00__GPIO3_IO05 	0x10b0 /* 69 DATA_AREADY */
			MX6UL_PAD_LCD_DATA01__GPIO3_IO06 	0x10b0 /* 70  HANDSHAKE_PIN */
		>;
	};
};

&ecspi1 {
	#address-cells = <1>;
	#size-cells = <0>;
	fsl,spi-num-chipselects = <1>;
    cs-gpios = <&gpio3 26 GPIO_ACTIVE_LOW>;
	pinctrl-names = "default";
	pinctrl-0 = <&pinctrl_ecspi1>;
	status = "okay";
};

&csi {
	status = "okay";
};

&ov5640 {
	status = "okay";
};

&lcdif {
	status = "disabled";
};

/delete-node/ &sim2;

**Comment by Oquirella** [comment]
&ecspi1 {
#address-cells = <1>;
#size-cells = <0>;
fsl,spi-num-chipselects = <1>;
cs-gpios = <&gpio3 26 GPIO_ACTIVE_LOW>;
pinctrl-names = "default";
pinctrl-0 = <&pinctrl_ecspi1>;
status = "okay";
};

Do in need to change esp_board.bus_num and esp_board.chip_select to 1 according to this device tree?

**Comment by mantriyogesh** [comment]
Nope. You clearly do not have first event from ESP.

First thing first, On loading above stated device tree, can you list your spi device in /dev?

```sh
$ ls -lrt /dev/spi*
```

**Comment by Oquirella** [comment]
I don't have a spi device registrered, it is not clear for me if i need to have it or not.

**Comment by Oquirella** [comment]
I also get the following spi error messages:

   24.521268] spi spi0.0: SPI transfer failed: -22
[   24.617034] spi_master spi0: failed to transfer one message from queue
[   24.752438] SPI Transaction failed: -22
[   25.477800] spi spi0.0: SPI transfer failed: -22
[   25.622187] spi_master spi0: failed to transfer one message from queue
[   25.628764] SPI Transaction failed: -22
[   25.659423] spi spi0.0: SPI transfer failed: -22
[   25.736373] spi_master spi0: failed to transfer one message from queue
[   25.850979] SPI Transaction failed: -22

**Comment by Oquirella** [comment]
GPIO pins

root@localhost:~# cat /sys/kernel/debug/gpio 
gpiochip0: GPIOs 0-31, parent: platform/209c000.gpio, 209c000.gpio:
 gpio-3   (                    |xnur                ) in  lo ACTIVE LOW
 gpio-9   (                    |regulator-sd1-vmmc  ) out lo 
 gpio-19  (                    |cd                  ) in  hi IRQ ACTIVE LOW

gpiochip1: GPIOs 32-63, parent: platform/20a0000.gpio, 20a0000.gpio:

gpiochip2: GPIOs 64-95, parent: platform/20a4000.gpio, 20a4000.gpio:
 gpio-69  (                    |SPI_DATA_READY_PIN  ) in  lo IRQ 
 gpio-70  (                    |SPI_HANDSHAKE_PIN   ) in  hi IRQ 
 gpio-90  (                    |spi_imx             ) out hi 

gpiochip3: GPIOs 96-127, parent: platform/20a8000.gpio, 20a8000.gpio:

gpiochip4: GPIOs 128-159, parent: platform/20ac000.gpio, 20ac000.gpio:
 gpio-132 (                    |headphone detect    ) in  lo IRQ 
 gpio-135 (                    |spi5 CS0            ) out hi ACTIVE LOW
 gpio-136 (                    |?                   ) out lo 
 gpio-138 (                    |mosi                ) out hi 
 gpio-139 (                    |sck                 ) out lo 

gpiochip5: GPIOs 504-511, parent: spi/spi5.0, 74hc595, can sleep:
 gpio-507 (                    |regulator-can-3v3   ) out hi ACTIVE LOW

**Comment by mantriyogesh** [comment]
It is getting a little distracted. I will try to understand step by step.

1. Device Tree
a. SPI blob
I think the first step is to get your device tree correct. 
Once you have correct dts, you should build it and load as dtb into your system (mostly in `/boot` but depends upon your platform).

Please **comment** out any **spidev disabler** blob (if any added by you) in device tree or overlay

Once reboot, it should load your dtb automatically. If the device tree is correct,  
```sh
ls -lrt /dev/spi*
```
should list something.

b. extra GPIOs 
- We will deal this later

Is 1.a step working for you?

**Comment by Oquirella** [comment]
I now got the spidev device

root@localhost:~# ls -l /dev/spidev0.0 
crw------- 1 root root 153, 0 Sep 20 10:44 /dev/spidev0.0

But i also get this error:

Failed Init SPI device

**Comment by mantriyogesh** [comment]
Great.

Now,
1. b) `GPIO configuration`
a. Save your Device tree config (dts) in some directory like /home/<user>
b. Configure the GPOIs for resetpin, handshake and data_ready. More info can be found in porting guide.
c. Possibly verify them that they are correctly configured from device tree.

2. `User-space tool testing of SPI`
a. Test your SPI with spidev or with any user-space library, which will make sure that your SPI is working as expected.
b. Alternatively, TX and RX both messages you should test with ESP chipset's SPI.
You can use https://github.com/espressif/esp-idf/tree/master/examples/peripherals/spi_slave for dummy transfer. 
You can also use exact SPI pins for your ESP chipsets. You can find them in [spi_slave_api.c](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/esp/esp_driver/network_adapter/main/spi_slave_api.c#L45)
This will make sure your All SPI pins and 'Handshake' are working fine.

Once you verify some back to back Tx & Rx SPI transactions, we will move to next step.

Do not hurry in trying ESP-Hosted directly. Let us get the base platform working first.

**Comment by Oquirella** [comment]
I managed to get the ethap0 & ethsta0 interfaces, but when i try to enable them i get the following error:

root@localhost:~# ifconfig ethap0 up
ifconfig: SIOCSIFFLAGS: Cannot assign requested address

**Comment by mantriyogesh** [comment]
can you please post dmesg?
Is your sta interface up?

**Comment by mantriyogesh** [comment]
what is output of `sudo ifconfig -a`? Is there MAC address for interface you intend to bring up?

**Comment by Oquirella** [comment]
dmesg:

[56187.838801] Received INIT event from ESP32 peripheral
[56187.850072] ESP Reconfigure SPI CLK to 30 MHz
[56188.086722] ESP32 peripheral is registered to SPI bus [0],chip select [0], SPI Clock [30]
[56188.334081] ESP peripheral capabilities: 0x68

ifconfig:
ethap0    Link encap:Ethernet  HWaddr 00:00:00:00:00:00  
          BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

ethsta0   Link encap:Ethernet  HWaddr 00:00:00:00:00:00  
          BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

**Comment by mantriyogesh** [comment]
Have you followed https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/docs/Linux_based_host/Getting_started.md and c demo app, https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/docs/common/c_demo.md docs?

If yes, What is the output of 
```sh
sudo ./test.out get_sta_mac_addr
sudo ./test.out get_ap_scan_list
```

**Comment by Oquirella** [comment]
Oke, i am very close now:


Linux (still problems, any idea why?):

root@localhost:~# esp-hosted get_ap_scan_list
Enable heartbeat with duration 20
Control response timed out after 30 sec
Response not received


ESP32:

I (2624) slave_ctrl: event ESPInit
I (7394) slave_ctrl: HB timer started for 20 sec

I (37394) slave_ctrl: Station mode set in scan handler
I (38994) slave_ctrl: Total APs scanned = 3
I (38994) slave_ctrl: Details of AP no 0
I (38994) slave_ctrl: 
SSID                    XXX
RSSI                    -57
Channel                 6
BSSID                   00:b7:71:44:29:60
Auth mode               5

I (39014) slave_ctrl: Details of AP no 1
I (39014) slave_ctrl: 
SSID                    XXXX
RSSI                    -57
Channel                 6
BSSID                   00:b7:71:44:29:61
Auth mode               0

I (39024) slave_ctrl: Details of AP no 2
I (39024) slave_ctrl: 
SSID                    iot_devices
RSSI                    -57
Channel                 6
BSSID                   00:b7:71:44:29:62
Auth mode               3

**Comment by mantriyogesh** [comment]
1. Can you please check if your Reset, Data ready and Handshake pins are working?
- `ResetPin`:
Every loading of esp32 driver should reset the ESP32. (you will see the logs are dumped because of reloading of firmware image). This is equally important to other pins.
- For confirming Handshake and data ready, you can add debug traces in functions in `ESP-Hosted_MCU_Host/host/linux/host_driver/esp32/spi/esp_spi.c`
  - `Handshake`: [spi_interrupt_handler](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/host/linux/host_driver/esp32/spi/esp_spi.c#L82)
  - `Data ready`: [spi_data_ready_interrupt_handler](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/host/linux/host_driver/esp32/spi/esp_spi.c#L73)

If these functions do not hit,

2. Try Smaller SPI CLK freq
- replace [spi_clk_mhz](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/host/linux/host_driver/esp32/spi/esp_spi.c#L354) in `ESP-Hosted_MCU_Host/host/linux/host_driver/esp32/spi/esp_spi.c` with say 1 (desired small freq in MHz)
(Next) SPI transaction will be triggered by SPI Master(host) once interrupt received over these two lines.

3. Try changing SPI mode at both sides
- Host
  - Change [SPI Mode](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/host/linux/host_driver/esp32/spi/esp_spi.c#L353) in `ESP-Hosted_MCU_Host/host/linux/host_driver/esp32/spi/esp_spi.c` to `SPI_MODE_3`
- ESP
  - Use same SPI mode  [spi_slave_interface_config_t slvcfg.mode](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/esp/esp_driver/network_adapter/main/spi_slave_api.c#L501) in `Hosted_MCU_Host/esp/esp_driver/network_adapter/main/spi_slave_api.c`
  
Yes! you are now much close. Your Host->ESP messages are working. Once ESP->Host messages work, your setup is complete.

**Comment by Oquirella** [comment]
My pin configuration looks correct:

gpiochip2: GPIOs 64-95, parent: platform/20a4000.gpio, 20a4000.gpio:
 gpio-67  (                    |sysfs               ) in  hi 
 gpio-69  (                    |SPI_DATA_READY_PIN  ) in  lo IRQ 
 gpio-70  (                    |SPI_HANDSHAKE_PIN   ) in  hi IRQ

I also tried to change the SPI mode to 3, but nothing worked anymore.

**Comment by mantriyogesh** [comment]
Can you independently test these GPIOs as this would be easier.

Possible reasons:
1. GPIOs engaged in other function/peripherals.
2. Device tree may be incorrect (mostly this is not your reason)
3. loose wires / very long wires. (For SPI, we recommend to use <10cm, all equal wires)

Possible debugging options:
1. It is also worth to try replacing these GPIO pins with other known free (unused) GPIO pins in device tree
2. Change wires
3. Check continuity on multimeter

**Comment by mantriyogesh** [comment]
@Oquirella Any update?

**Comment by Oquirella** [comment]
I reconfigured the pins but i still don't get any reponse. I get the HANDSHAKE & DATA READY events:

root@localhost:~# esp-hosted get_ap_scan_list
Enable heartbeat with duration 20
[  192.796974] HAND SHAKE
[  192.799836] DATA READY
[  192.804471] HAND SHAKE
[  192.808280] HAND SHAKE
[  212.695066] DATA READY
[  212.696897] HAND SHAKE
[  212.701082] HAND SHAKE
Control response timed out after 30 sec
Response not received

**Comment by mantriyogesh** [comment]
Can you please 
1. Confirm if your ESP and host are using exact same git commit?
2. Send current complete ESP log, dmesg log and test.out run log
3. Do you have logic analyser?

**Comment by mantriyogesh** [comment]
4. Have you also tried SPI mode 1 at both places?

5. SPI mode mismatch can also be tried to understand further.
For example, Host at SPI Mode3, and ESP at SPI Mode0.
If these combination works, This definitely means timing issues.

**Comment by Oquirella** [comment]
<6>[   96.767388] Received INIT event from ESP32 peripheral
<6>[   96.767402] EVENT: 2
<6>[   96.767407] EVENT: 1
<6>[   96.767413] ESP Reconfigure SPI CLK to 30 MHz
<6>[   96.980102] ESP32 peripheral is registered to SPI bus [0],chip select [0], SPI Clock [30]
<6>[   97.199450] EVENT: 0
<6>[   97.199463] ESP peripheral capabilities: 0x68
<6>[   97.441310] HAND SHAKE
<6>[   97.445951] DATA READY
<6>[   97.447408] HAND SHAKE
<6>[   97.449026] HAND SHAKE
<6>[   97.776185] DATA READY
<6>[   97.778322] HAND SHAKE

I can receive some events, so why can't i receive the other stuff?

**Comment by mantriyogesh** [comment]
Hello @Oquirella, Actually this might need some level of debugging.

1. From your earlier logs, MISO & MOSI both had worked fine.
So, can you please just comment [adjust_spi_clock](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/host/linux/host_driver/esp32/spi/esp_spi.c#L185) to keep initial 10MHz freq as is without change?
2. Overall, Can you try to bypass data ready (like assuming it is always on)
Just change 
- a. [rx_pending](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/host/linux/host_driver/esp32/spi/esp_spi.c#L269) = 1;           in `ESP-Hosted_MCU_Host/host/linux/host_driver/esp32/spi/esp_spi.c`-> _esp_spi_work esp_spi_work()_
- b. comment 
  ```sh
  //WRITE_PERI_REG(GPIO_OUT_W1TC_REG, (1 << gpio_data_ready));
  ``` 
  in [ESP-Hosted_MCU_Host/esp/esp_driver/network_adapter/main/spi_slave_api.c](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/esp/esp_driver/network_adapter/main/spi_slave_api.c#L315) -> `get_next_tx_buffer()`
3. If (1) & (2) doesn't work, then I will send you debug patch, which might let us know what is the problem.

**Comment by Oquirella** [comment]
It finally works now, keeping the initial 10MHz did the trick.

**Comment by mantriyogesh** [comment]
But your frequency bumped to 30MHz. So you must be using C3/S2/S3. Is that correct?

**Comment by Oquirella** [comment]
I am using S3

**Comment by mantriyogesh** [comment]
It is good to know that it is working for you.

However, 10MHz looks less.
Can you further optimize frequency by increasing in 1 MHz step from 10MHz?
You can change [SPI_CLK_MHZ](https://github.com/espressif/esp-hosted/blob/ESP-Hosted_MCU_Host/esp/esp_driver/network_adapter/main/spi_slave_api.c#L111) .

The overall throughput or data rate will be dependent upon the SPI CLK is released from host. ESP32-S3 supports [upto 60MHz](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-reference/peripherals/spi_slave.html) SPI slave driver clock. We have tested till 30MHz with Raspberry-Pi.

Some of the things to verify:
1. Is correct input power is reaching to your SoC (hence the peripherals)
2. Try smaller jumper wires / PCBs
3. Max supported SPI clock frequency in data sheet
4. Other than this, mostly would be related to timings of SPI at higher clock. In such case, we can help you further with Logic analyzer capture.

**Comment by KashyapPatalia** [comment]
Hello, everyone. I am getting following logs for dmesg (I am using Raspberry Pi 4 model B with 32 bit Raspberry Pi OS Lite + ESP32_Devkitc_v4 with pre built binaries of spi v0.4 - SPI connection):

[   94.113919] OF: overlay: WARNING: memory leak will occur if overlay removed, property: /soc/spi@7e204000/spidev@0/status
[   94.243688] Bluetooth: Core ver 2.22
[   94.243757] NET: Registered PF_BLUETOOTH protocol family
[   94.243763] Bluetooth: HCI device and connection manager initialized
[   94.244099] Bluetooth: HCI socket layer initialized
[   94.244109] Bluetooth: L2CAP socket layer initialized
[   94.244125] Bluetooth: SCO socket layer initialized
[  116.410158] esp32_spi: loading out-of-tree module taints kernel.
[  116.410925] esp_reset, ESP32: Resetpin of Host is 6
[  116.411147] esp_reset, ESP32: Triggering ESP reset.
[  116.411536] ESP32 peripheral is registered to SPI bus [0],chip select [0], SPI Clock [10]
[  118.822285] 
               Received INIT event from ESP32 peripheral
[  118.822306] EVENT: 2
[  118.822323] EVENT: 1
[  118.822337] Unsupported tag in event
[  118.822350] EVENT: 0
[  118.822364] ESP peripheral capabilities: 0x78
[  119.575877] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[  119.575890] Bluetooth: BNEP filters: protocol multicast
[  119.575902] Bluetooth: BNEP socket layer initialized
[  119.588172] NET: Registered PF_ALG protocol family
[  119.608597] cryptd: max_cpu_qlen set to 1000

I do not know this _**[  118.822337] Unsupported tag in event**_. Also I am not able to establish communication over SPI. It always says timeout & Response not received.
Can anyone please help me ?

**Comment by mantriyogesh** [comment]
We can certainly help you @KashyapPatalia .
Can you please raise a separate issue for this?

**Comment by KashyapPatalia** [comment]
> Thanks for the quick response. Sure, let me raise a separate issue for this.

**Comment by meera1310** [comment]
> 4. Have you also tried SPI mode 1 at both places?
> 
>     5. SPI mode mismatch can also be tried to understand further.
>        For example, Host at SPI Mode3, and ESP at SPI Mode0.
>        If these combination works, This definitely means timing issues.

How to set spi mode in esp32 side.

**Comment by meera1310** [comment]
> I am very close.
> 
> dmesg:
> 
> [56187.838801] Received INIT event from ESP32 peripheral [56187.838831] EVENT: 2 [56187.847831] EVENT: 1 [56187.850072] ESP Reconfigure SPI CLK to 30 MHz [56188.086722] ESP32 peripheral is registered to SPI bus [0],chip select [0], SPI Clock [30] [56188.331843] EVENT: 0 [56188.334081] ESP peripheral capabilities: 0x68 [51534.518133] Bus freq set to 400000000 start... [51534.518574] Bus freq set to 400000000 done! [51537.842456] [51537.842456] Bus freq set to 24000000 start... [51537.842824] Bus freq set to 24000000 done! [51537.878057] [51537.878057] Bus freq set to 400000000 start... [51537.878481] Bus freq set to 400000000 done! [51544.002397] [51544.002397] Bus freq set to 24000000 start... [51544.002767] Bus freq set to 24000000 done! [51544.038028] [51544.038028] Bus freq set to 400000000 start... [51544.038466] Bus freq set to 400000000 done! [51549.122407] [51549.122407] Bus freq set to 24000000 start... [51549.122787] Bus freq set to 24000000 done! [51549.158021] [51549.158021] Bus freq set to 400000000 start... [51549.158449] Bus freq set to 400000000 done!
> 
> ifconfig: ethap0 Link encap:Ethernet HWaddr 00:00:00:00:00:00 BROADCAST MULTICAST MTU:1500 Metric:1 RX packets:0 errors:0 dropped:0 overruns:0 frame:0 TX packets:0 errors:0 dropped:0 overruns:0 carrier:0 collisions:0 txqueuelen:1000 RX bytes:0 (0.0 B) TX bytes:0 (0.0 B)
> 
> ethsta0 Link encap:Ethernet HWaddr 00:00:00:00:00:00 BROADCAST MULTICAST MTU:1500 Metric:1 RX packets:0 errors:0 dropped:0 overruns:0 frame:0 TX packets:0 errors:0 dropped:0 overruns:0 carrier:0 collisions:0 txqueuelen:1000 RX bytes:0 (0.0 B) TX bytes:0 (0.0 B)

How did you get this INT messages? and what did you do for spidev_disabler in yocto. Please help me

**Comment by mantriyogesh** [comment]
@Oquirella Can you please help to close this issue?
